redash/handlers/query_results.py

```diff
@@ -108,18 +108,23 @@ def post(self):
 
 class QueryResultDropdownResource(BaseResource):
     def get(self, query_id):
+        query = get_object_or_404(models.Query.get_by_id_and_org, query_id, self.current_org)
+        require_access(query.data_source, current_user, view_only)
         return dropdown_values(query_id)
 
 
 class QueryDropdownsResource(BaseResource):
     def get(self, query_id, dropdown_query_id):
         query = get_object_or_404(models.Query.get_by_id_and_org, query_id, self.current_org)
+        require_access(query, current_user, view_only)
 
         related_queries_ids = [p['queryId'] for p in query.parameters if p['type'] == 'query']
         if int(dropdown_query_id) not in related_queries_ids:
-            abort(403)
+            dropdown_query = get_object_or_404(models.Query.get_by_id_and_org, dropdown_query_id, self.current_org)
+            if not has_access(dropdown_query.data_source, current_user, view_only):
```

Comment:
Can't we use `require_access` here as well?

Reply:
Totally. 13e398b